bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiRpc.java

```diff
@@ -666,19 +666,21 @@ public void onNotifyEvent(Shelly2RpcNotifyEvent message) {
 
                     case SHELLY2_EVENT_OTASTART:
                         logger.debug("{}: Firmware update started: {}", thingName, getString(e.msg));
-                        getThing().postEvent(e.event, true);
-                        getThing().setThingOffline(ThingStatusDetail.FIRMWARE_UPDATING,
+                        getThing().setThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.FIRMWARE_UPDATING,
```

Comment:
Why is this change necessary? Can the more dedicated method `setThingOffline` not be used?

If it's because you don't want these lines executed from that method in this scenario:
```java
api.close(); // Gen2: disconnect WS/close http sessions
watchdog = 0;
channelsCreated = false;
```

I think some refactoring is needed, because it's now quite error-prone and undocumented that you in some cases should use `setThingStatus(ThingStatus.OFFLINE)` and in other cases `setThingOffline`. I'm just guessing here, so would appreciate your feedback.

Reply:
Es, the difference is that setThingOffline() diaconnect from api/WebSocket, where setThingStatus() just updates the thing status. This makes sense, because firmware updates are a flow of multiple events, so I keep the WebSocket connection to receive further events.

I don't see an issue, maybe renaming setThinOffline to setOfflineAndDisconnect() makes it clearer?

Reply:
> I don't see an issue, maybe renaming setThinOffline to setOfflineAndDisconnect() makes it clearer?

Yes, that would clarify the difference and possibly avoid doubt/confusion/mistakes.

Reply:
maybe even better:
Due to the fact that status is now always OFFLINE and only the detail differ I could add a parameter to setThingOffline(detail, reason, disconnect). So setting status and detail is always the same and the caller specifies if the socket should be disconnected nor not.

Reply:
The method would (still) be responsible for doing two different things, and the naming only suggests one of them. However, since it was like this before, I won't insist, although adding a boolean parameter branches the method into two different paths.

You could consider creating a separate method, `disconnect`, that could be called in addition to setting the Thing status to **OFFLINE**.

Reply:
Changed it to setThingOfflineAndDisconnect()